cas-management-webapp-support/src/main/java/org/apereo/cas/mgmt/services/web/beans/RegisteredServiceEditBean.java

```diff
@@ -0,0 +1,302 @@
+        }
+
+        public void setAvailableAttributes(final List<String> availableAttributes) {
+            this.availableAttributes = availableAttributes;
+        }
+
+        /**
+         * Visible for serialization only. Use {@link FormData#getCustomComponent(String)} instead.
+         *
+         * @return all the custom components
+         */
+        public Map<String, Map<String, ?>> getCustomComponent() {
+            return this.customComponent;
+        }
+
+        /**
+         * Get the current properties for the specified custom component. The returned {@link Map} should only contain
+         * nested Maps, Arrays, and simple objects.
+         *
+         * @param componentName name of the component to get the properties for (this should be unique for each
+         *                      component)
+         * @return current custom component properties
+         */
+        public Map<String, ?> getCustomComponent(final String componentName) {
+            return this.customComponent.get(componentName);
+        }
+
+        /**
+         * This is reserved for usage by any custom components that need to present their config to the management UI.
+         * The provided {@link Map} should only contain nested Maps, Arrays, and simple objects.
+         *
+         * @param componentName name of the component to store the properties for (this should be unique for each
+         *                      component)
+         * @param properties    custom component properties
+         */
+        public void setCustomComponent(final String componentName, final Map<String, ?> properties) {
+            this.customComponent.put(componentName, properties);
+        }
+    }
+
+    /**
+     * The type Service data.
+     */
+    public static class ServiceData {
+        private String assignedId;
+        private String serviceId;
+        private String name;
+        private String description;
+        private String logoUrl;
+        private String theme;
+        private int evalOrder = Integer.MIN_VALUE;
+        private Set<String> requiredHandlers = new HashSet<>();
+        private String logoutUrl;
+        private RegisteredServiceSupportAccessEditBean supportAccess = new RegisteredServiceSupportAccessEditBean();
+        private String type = RegisteredServiceTypeEditBean.CAS.toString();
+        private RegisteredServiceOAuthTypeEditBean oauth = new RegisteredServiceOAuthTypeEditBean();
+        private String logoutType = RegisteredServiceLogoutTypeEditBean.BACK.toString();
+        private RegisteredServiceUsernameAttributeProviderEditBean userAttrProvider =
+                new RegisteredServiceUsernameAttributeProviderEditBean();
+        private RegisteredServicePublicKeyEditBean publicKey = new RegisteredServicePublicKeyEditBean();
+        private RegisteredServiceProxyPolicyBean proxyPolicy = new RegisteredServiceProxyPolicyBean();
+        private RegisteredServiceAttributeReleasePolicyEditBean attrRelease
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['VARIABLE_DEF' should be separated from previous statement.](https://www.codacy.com/app/mmoayyed/cas/file/2523380941/issues/source?bid=3256115&fileBranchId=3280886#l120)